src/main/java/utils/command/AddCardToTagCommand.java

```diff
@@ -8,26 +8,37 @@
 import model.TagList;
 import model.TagUUID;
 import utils.UserInterface;
+import utils.exceptions.InkaException;
+import utils.exceptions.UUIDWrongFormatException;
 import utils.storage.IDataStorage;
 
 public class AddCardToTagCommand extends Command {
     private String tagName;
     private TagUUID tagUUID;
     private CardUUID cardUUID;
 
-    public AddCardToTagCommand(String tagName, String cardUUID) {
+    public AddCardToTagCommand(String tagName, String cardUUID) throws InkaException {
         this.tagName = tagName;
-        this.cardUUID = new CardUUID(UUID.fromString(cardUUID));
+        try {
+            this.cardUUID = new CardUUID(UUID.fromString(cardUUID));
+        } catch (IllegalArgumentException e) {
+            throw new UUIDWrongFormatException();
+        }
     }
 
-    private void addCardToTag(TagList tagList, CardList cardList, UserInterface ui) {
+    /**
+     * Adds the tagName tag to the Card with cardUUID, creates the tag beforehand if it currently does not exist.
+     *
+     * @param tagList  The tagList into which to add the tag that has been created
+     * @param cardList The cardList from which to find the card with cardUUID
+     * @param ui       The userInterface to print the success of the tag creation
+     * @throws InkaException
+     */
+    private void addCardToTag(TagList tagList, CardList cardList, UserInterface ui) throws InkaException {
         //find the corresponding Tag and Card based on its tagName and card uuid
         Tag tagToAdd = tagList.findTagFromName(tagName);
         Card cardToAdd = cardList.findCardFromUUID(cardUUID);
-
-        if (cardToAdd == null) {
-            //means the card doesn't exist yet, throw exception
-        }
+        assert cardToAdd != null;
```

Comment:
If `cardToAdd` is null this will crash the whole program, I think it's better to throw an exception here

Reply:
Oh I am already throwing an exception in `cardList.findCardFromUUID(cardUUID)`, the assertion is just another layer of protection to make sure that it will never be null just like the whole point of having an assertion @ltzehan

Reply:
Ok fair enough 👍